linalg/vector.cpp

```diff
@@ -175,6 +175,15 @@ Vector &Vector::operator-=(const Vector &v)
    return *this;
 }
 
+Vector &Vector::operator+=(double c)
+{
+   const bool use_dev = UseDevice();
+   const int N = size;
+   auto y = ReadWrite(use_dev);
+   MFEM_FORALL_SWITCH(use_dev, i, N, y[i] += c;);
```

Comment:
I might consider changing the implementation of `operator-=(double)` to be `(*this) += -c` to reduce code duplication. Something similar IMO should be done in the implementation of `operator/=` but that might not belong in this PR.

Reply:
As far as I know, `a /= b;` is different (more accurate?) from `a *= (1./b);` so in some cases you may want to explicitly use `/=`. I'm not sure if there is some subtle difference between `a -= b;` and `a += (-b);`, so it may be better to keep both kernels.

Reply:
MFEM's `Vector::operator/=` implementation actually uses `y[i] *= (1./c)` so there's really no difference in that case. But in general I don't care that much, I am happy to see this PR go in either way.

Reply:
Good point about the current implementation of `Vector::operator/=`. I think this optimization can be useful, since, as far as I know, on most platforms division is several times more expensive than multiplication. In any case, I don't think we have an immediate need to make changes to these methods.

Reply:
Another possibility for factoring the code could be something like this:

```
template <typename Op>
Vector &Vector::componentwise_scalar_op(double c, Op op)
{
  const bool use_dev = UseDevice();
  const int N = size;
  auto y = ReadWrite(use_dev);
  MFEM_FORALL_SWITCH(use_dev, i, N, y[i] = op(y[i],c); );
  return *this;
}
  
Vector &Vector::operator+=(double c)
{
  componentwise_scalar_op<std::plus<double>>(c);
}
  
Vector &Vector::operator-=(double c)
{
  componentwise_scalar_op<std::minus<double>>(c);
}

Vector &Vector::operator*=(double c)
{
  componentwise_scalar_op<std::multiplies<double>>(c);
}
  
Vector &Vector::operator/=(double c)
{
  componentwise_scalar_op<std::divides<double>>(c);
}

```

Reply:
I made a separate issue and PR for this as a proposed refactoring.  So it can wait for after this PR.

Reply:
I don't have any meaningfull opinion on this! I simple added the function I needed. I was so out of my depth with all the kernel stuff I added a unit test, just to make sure. @rw-anderson your code looks better in terms of maintainance/extensions, as only one functions seems to be affected.